### PR TITLE
chore(ui-components): align linting and fix issues

### PR DESCRIPTION
**Pull request by Klaus-Keller**
This pull requests aligns eslint configuration and fixes eslint reported issues for module `@sap-ux/ui-components`.

What I did in this pull request:

- 🍒-picked @Jimmy-Joseph19 's commit https://github.com/SAP/open-ux-tools/pull/689/commits/86189e40e6686aab9a595830d100ba1d9ccdc7b2 which contains lint fixes for module `@sap-ux/ui-components`.
- ~~Resolved conflicting eslint rule `quoteProps` https://github.com/SAP/open-ux-tools/commit/4fd115cd26cfb2e60f1aa639e233e11026af18e8, for further details see: https://github.com/SAP/open-ux-tools/pull/689#issuecomment-1255628784 and https://github.com/SAP/open-ux-tools/pull/689#issuecomment-1256112736~~ **->** was fixed in PR https://github.com/SAP/open-ux-tools/pull/701
- adjusted local settings in `packages/ui-components/.eslintrc`. Basically, I've removed rules to fall back to the root eslint settings and had to add one rule: `"react/no-unknown-property": ["error", { "ignore": ["onFocusCapture"] }]`. Problem here is, that react 16 supports property `onFocusCapture`, but `eslint-plugin-react@7.31.6` doesn't like it, so I've added it to ignore list: https://github.com/SAP/open-ux-tools/commit/72e8466c280aa3c1be0a862e56cdba54756e40b1
- deleted all occurrences of comment `eslint-disable` and fixed the eslint issues in sources: https://github.com/SAP/open-ux-tools/commit/0fa1d3be72b2a50d9376ce613b70ac7e85d421a9 and in tests: https://github.com/SAP/open-ux-tools/commit/673285013359d2b9f77d6c2b932daaa0b225536c. I've started with `pnpm lint:fix:all` which updated many JSDoc comments automatically and continued with manual fixes.

There are three issues left:
```
open-ux-tools/packages/ui-components/test/unit/components/UICombobox.test.tsx
  214:57  warning  Too many nested callbacks (4). Maximum allowed is 3  max-nested-callbacks
  277:57  warning  Too many nested callbacks (4). Maximum allowed is 3  max-nested-callbacks

open-ux-tools/packages/ui-components/test/unit/components/UIFlexibleTable.test.tsx
  706:30  warning  Too many nested callbacks (4). Maximum allowed is 3  max-nested-callbacks

✖ 3 problems (0 errors, 3 warnings)
```

This will vanish once we've updated the ruleset for linting of tests. See discussion here: https://github.com/SAP/open-ux-tools/pull/689#discussion_r978089651

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -255,7 +255,8 @@
         "prettier/prettier": [
             "error",
             {
-                "endOfLine": "auto"
+                "endOfLine": "auto",
+                "quoteProps": "preserve"
```
Comment (tobiasqueck) [inline .eslintrc:259]: This PR shouldn't change central settings and only focus on the `ui-components` module. The central changes are reserved for PR #689

**Review by tobiasqueck — CHANGES_REQUESTED** [review]

File: .eslintrc (hunk 1 above)
Comment (Klaus-Keller) [reply .eslintrc:259]: @tobiasqueck, problem is, we have conflicting eslint settings in `main`-branch, which means I can't solve the eslint issues in `ui-components` without resolving this conflict first. For background information about the issue with conflicting rules see https://github.com/SAP/open-ux-tools/pull/689#issuecomment-1255628784 and https://github.com/SAP/open-ux-tools/pull/689#issuecomment-1256112736.

If we do not fix it here, we need either to add a `eslint-disable` comment to the code and remove it after the change has done, or, which I think is the better approach, I can create a separate PR with just the fix for conflicting eslint rules.

Comment (Klaus-Keller) [reply .eslintrc:259]: -> https://github.com/SAP/open-ux-tools/pull/701

**Review by tobiasqueck — APPROVED** [review]
Great improvement of the code quality, and I couldn't spot any accidental regressions.
In detail: 
* thanks for removing a lot of deactivated rules in the `ui-components/.eslintrc`
* all the changes to jsdoc comments looks good
* thanks for using better function names e.g. `getUIId` 
* the consistent handling of return values for every path make the code more readable
* the remove of `eslint-disable` sections is also very nice

**Review by devinea — APPROVED** [review]
Changes look good. Thanks

**Pull request merged**